quickwit-index-config/src/config.rs

```diff
@@ -70,14 +80,51 @@ pub enum SortBy {
     DocId,
 }
 
-/// Convert a field (name, value) into a tag string `name:value`.
+/// Converts a field (name, value) into a tag string `name:value`.
 pub fn convert_tag_to_string(field_name: &str, field_value: &Value) -> String {
-    format!("{}:{}", field_name, tantivy_value_to_string(field_value))
+    let field_value_str = tantivy_value_to_string(field_value);
+    if field_value_str == MANY_TAG_VALUES {
+        return format!(
+            "{}{}{}{}",
+            field_name, TAG_FIELD_VALUE_SEPARATOR, TAGS_VALUE_ESCAPE, field_value_str
+        );
+    }
+    format!(
+        "{}{}{}",
+        field_name, TAG_FIELD_VALUE_SEPARATOR, field_value_str
+    )
 }
 
 /// Returns true if tag_string is of form `{field_name}:any_value`.
 pub fn match_tag_field_name(field_name: &str, tag_string: &str) -> bool {
-    tag_string.starts_with(&format!("{}:", field_name))
+    tag_string.starts_with(&format!("{}{}", field_name, TAG_FIELD_VALUE_SEPARATOR))
+}
+
+/// Extracts the field name from a tag value of the form `{field_name}:any_value`.
+pub fn extract_field_name_from_tag_value(tag_value: &str) -> Option<String> {
+    tag_value
+        .split_once(TAG_FIELD_VALUE_SEPARATOR)
+        .map(|(field_name, _)| field_name.to_string())
+}
+
+/// Escapes the tag value by prefixing the value part of this format `{field_name}:any_value`
+/// by `\` if `any_value` is the wildcard value `*`.
+pub fn escape_tag_value(tag_value: &str) -> String {
+    match tag_value.split_once(TAG_FIELD_VALUE_SEPARATOR) {
```

Comment:
Yes, but we wanted to keep this wildcard as const so to change it in case we favored `_too_many_value`. This is why I am using all these variable interpolations

Reply:
Sorry my comment was not accurate at all! I will rewrite it.